src/main/kotlin/no/nb/mlt/wls/domain/model/Item.kt

```diff
@@ -11,4 +15,32 @@ data class Item(
     val callbackUrl: String?,
     val location: String?,
     val quantity: Int?
-)
+) {
+    fun pickItem(amountPicked: Int): Item {
+        val itemsInStockQuantity = quantity ?: 0
+
+        // In the case of over-picking, log it and set quantity to zero.
+        // This is in hope that on return the database recovers
+        if (amountPicked > itemsInStockQuantity) {
+            logger.error {
+                "Tried to pick too many items for item with id '$hostId'. " +
+                    "WLS DB has $itemsInStockQuantity stocked, and storage system tried to pick $amountPicked"
+            }
+        }
+        val quantity = Math.clamp(itemsInStockQuantity.minus(amountPicked).toLong(), 0, Int.MAX_VALUE)
+
+        val location: String =
+            if (quantity == 0) {
+                "WITH_LENDER"
+            } else if (location != null) {
+                location
+            } else {
+                // Rare edge case. Log it until we can determine if this actually happens in production
+                logger.error {
+                    "Item with ID $hostId for host $hostName without a location was picked. Location was set to empty string."
+                }
+                ""
```

Comment:
Maybe set this to something more meaningful, like "UNKNOWN"? Since "" can happen by accident, whereas if location is "UNKNOWN" then we are both sure that it was intentionally set, and where it came from, instead of it being a mistake that someone left out an empty string.
I might also be misremembering, but I think Item's validation function throws if location is set to an empty string? Or at least I feel like it should.

Reply:
```suggestion
                "UNKNOWN"
```